05_stringer/nicholascross/main_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := "127.0.0.1"
+	actual := buf.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected, actual)
+	}
+}
+
+func TestIPAddr_String(t *testing.T) {
+	testCases := map[string]struct {
+		input    IPAddr
+		expected string
+	}{
+		"localhost": {input: IPAddr{127, 0, 0, 1}, expected: "127.0.0.1"},
+		"gateway":   {input: IPAddr{192, 168, 0, 1}, expected: "192.168.0.1"},
+		"google":    {input: IPAddr{172, 217, 25, 142}, expected: "172.217.25.142"},
```

Comment:
Can you add a test for `IPAddr` with some more edge cases
`IPAddr{192, 163}`
`IPAddr{}`
Surprisingly, these are valid even though `IPAddr` enforces an array of size 4. Think about what these will initialize to.

Reply:
Oh right, yeah good point, totally didn't consider zero values! 👍